File: onigokko/src/main/java/com/reline/tag/injection/component/PresenterComponent.java
package com.reline.tag.injection.component;

import com.reline.tag.injection.module.NetworkModule;
import com.reline.tag.login.LoginPresenter;

import dagger.Component;

@Component(modules = NetworkModule.class, dependencies = ApplicationComponent.class)
public interface PresenterComponent {
    void inject(LoginPresenter presenter);
}

File: onigokko/src/main/java/com/reline/tag/injection/component/PresenterComponent.kt
package com.reline.tag.injection.component

import com.reline.tag.injection.module.NetworkModule
import com.reline.tag.login.LoginPresenter
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(NetworkModule::class))
interface PresenterComponent {
    fun inject(presenter: LoginPresenter)
}